components/toast_wrapper/toast_wrapper.jsx

```diff
@@ -229,45 +230,50 @@ class ToastWrapper extends React.PureComponent {
     }
 
     render() {
+        const {atLatestPost, atBottom, width, lastViewedAt, children} = this.props;
+        const {showUnreadToast, showNewMessagesToast, showMessageHistoryToast, unreadCount} = this.state;
+        const toastPresent = showUnreadToast || showNewMessagesToast || showMessageHistoryToast;
+
         let unreadToastProps = {
             show: false,
-            width: this.props.width,
+            width,
         };
 
         const archiveToastProps = {
-            show: this.state.showMessageHistoryToast,
-            width: this.props.width,
+            show: showMessageHistoryToast,
+            width,
             onDismiss: this.hideArchiveToast,
             onClick: this.scrollToLatestMessages,
             onClickMessage: Utils.localizeMessage('postlist.toast.scrollToBottom', 'Jump to recents'),
             showActions: true,
             extraClasses: 'toast__history',
         };
 
-        if (this.state.showUnreadToast && this.state.unreadCount > 0) {
+        if (showUnreadToast && unreadCount > 0) {
             unreadToastProps = {
                 ...unreadToastProps,
                 onDismiss: this.hideUnreadToast,
                 onClick: this.scrollToLatestMessages,
                 onClickMessage: Utils.localizeMessage('postlist.toast.scrollToBottom', 'Jump to recents'),
                 show: true,
-                showActions: !this.props.atLatestPost || (this.props.atLatestPost && !this.props.atBottom),
+                showActions: !atLatestPost || (atLatestPost && !atBottom),
```

Comment:
1/5 can be simplified to !(atLatestPost && atBottom)

Reply:
Aah, Nice. i cannot wrap my head around conditions usually. Will change